cmake/sca/gcc/sca.cmake

```diff
@@ -1,5 +1,43 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 # Copyright (c) 2024 Intel Corporation
+# Copyright (c) 2025 Alex Fabre
 
+# Get GCC version
+execute_process(
+  COMMAND ${CMAKE_C_COMPILER} -dumpfullversion
+  OUTPUT_VARIABLE GCC_COMPILER_VERSION
+  OUTPUT_STRIP_TRAILING_WHITESPACE
+  )
```

Comment:
The intention with the comment in https://github.com/zephyrproject-rtos/zephyr/pull/86290#discussion_r1971496939
was to remove this code completely, and instead have the compiler specific target code in https://github.com/zephyrproject-rtos/zephyr/blob/bc3a01a7e5c73a778b119717b56f673faa1e06b0/cmake/compiler/gcc/target.cmake#L42

This way the knowledge of how to obtain GCC version only needs to be maintained at one location (the compiler infrastructure).

To make things easy I made a commit with the intended changes here:
https://github.com/tejlmand/zephyr/commit/8af2e94708b13b3e54641e4c8fde820e8df95927

you may include https://github.com/tejlmand/zephyr/commit/8af2e94708b13b3e54641e4c8fde820e8df95927 in this PR.

```suggestion
```